python/pyarrow/table.pxi

```diff
@@ -565,6 +565,10 @@ cdef class RecordBatch:
         self.batch = NULL
         self._schema = None
 
+    def __init__(self):
+        raise RuntimeError("Do not call RecordBatch's constructor directly, "
```

Comment:
I would raise `TypeError`, not `RuntimeError`.

Reply:
Ok